src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2195,6 +2195,44 @@ exprt c_typecheck_baset::do_special_functions(
 
     return bswap_expr;
   }
+  else if(identifier=="__builtin_fpclassify" ||
+          identifier==CPROVER_PREFIX "fpclassify")
+  {
+    if(expr.arguments().size()!=6)
+    {
+      err_location(f_op);
+      error() << identifier << " expects six arguments" << eom;
+      throw 0;
+    }
+
+    // This gets 5 integers followed by a float or double.
+    // The five integers are the return values for the cases
+    // FP_NAN, FP_INFINITE, FP_NORMAL, FP_SUBNORMAL and FP_ZERO.
+    // gcc expects this to be able to produce compile-time constants.
+
+    if(expr.arguments()[5].type().id()!=ID_floatbv)
```

Comment:
How about `const exprt &input_value = arguments[5];` as that particular argument is used repeatedly?